docs/block-processing/us.md

```diff
@@ -0,0 +1,359 @@
+the stakeholder issues block like it has been done by the first stakeholder.
+Also the second one issues one more block as leader, so it produced 2 blocks.
+
+Then 4th epoch starts and sum of issuers stakes is `0.25 + 0.26 = 0.51 < 0.6` 
+(we take into account the second stakeholder only once). Adopting threshold is dumped to the minimal value `0.6`.
+So last adopted block version is still `0.1.0`, `lastKnownBlockVersion` is `0.2.0` for 
+the first and the second stakeholders and `0.1.0` for the third and the fourth.
+
+Then the third stakeholder downloads this update and issues block in the same manner like previous ones.
+
+When 5th epoch starts the sum of issuers stakes is `0.25 + 0.26 + 0.1 = 0.61 > 0.6` and 
+it's time to bump last adopted block version.
+So last adoped version is `0.2.0` now and `lastKnownBlockVersion` is `0.1.0` 
+for the latter stakeholder and `0.2.0` for others.
+
+When it's turn to issue block of the fourth stakeholder, 
+his software checks that his `lastKnownBlockVersion` mismatchs the last adopted block version
+and the software gives up issuing a block. So this slot left without a block.
+
+You can see despite the provided update system is powerful, it's also very complicated 
+and updates should be handled carefully.
+
+## Verification
+
+### Block header verification
+
+First of all, we check that block version of a verifiable block is _adopted_ or _competing_.
+
+### Update proposal verification
+
+#### General checks
+
+* _No duplicated proposer check_: that no one stakeholder sent two update proposals within current epoch. 
+We prohibit it because of opportunity spam update proposals from one stakeholder.
+
+* _Update proposal size check_: proposal is not exceed maximal proposal size.
+  * Update proposal size is computed as number of bytes in serialized `UpdateProposal`.
+  * Update proposal size limit is part of `BlockVersionData`.
+  * If the size of a proposal is greater than the limit from the adopted `BlockVersionData`, the proposal is invalid.
+
+* _Attributes known check_: if `verifyAllIsKnown` (defined [here](overall.md#verifyallisknown-flag))
+is `True`, all update proposal attributes must be known.
+ 
+* _Uniqueness of proposal check_: that there is no active proposal with the same id.
+
+* _Software version check_: numeric software version of application is 1 more 
+than of last confirmed proposal for this application.
+
+* _Stake to include into block check_: sum of positive votes for this proposal is enough 
+to be included into block (at least `updateProposalThd`).
+We wouldn't like adversary stakeholders to have opportunity to spam blockchain by `UpdateProposal`.
+So everyone can send update proposal to network, but it can get into block only along with enough votes.
+
+#### Protocol version checks
+
+* _Protocol version following check_: protocol version from proposal can follow last adopted protocol version.  
```

Comment:
I think it's better to move these rules above, to the place where you define **competing** block version. The word **competing** is used few times above, but the reader will know its actual meaning only after these lines.

Reply:
why `must`? we check that proposed protocol version can follow last adopted protocol version